src/Event/Decorator/SubjectFilterDecorator.php

```diff
@@ -0,0 +1,55 @@
+<?php
+/**
+ * CakePHP : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP Project
+ * @since         3.3.0
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Event\Decorator;
+
+use Cake\Event\Event;
+use RuntimeException;
+
+/**
+ * Event Subject Filter Decorator
+ */
+class SubjectFilterDecorator extends BaseDecorator
```

Comment:
Especially in pure listeners that are not behaviours, that deal with table objects, knowing the subject is pretty useful.